src/AbstractEditableRepository.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository;
+
+use Puli\Repository\Api\ChangeStream\ChangeStream;
+use Puli\Repository\Api\EditableRepository;
+use Puli\Repository\Api\Resource\PuliResource;
+use Puli\Repository\Api\UnsupportedOperationException;
+
+/**
+ * Abstract base for editable repositories providing tools to avoid code duplication.
+ *
+ * @since  1.0
+ *
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+abstract class AbstractEditableRepository extends AbstractRepository implements EditableRepository
+{
+    /**
+     * @var ChangeStream
+     */
+    private $changeStream;
+
+    /**
+     * Create the repository.
+     *
+     * @param ChangeStream|null $changeStream If provided, the repository will log
+     *                                        resources changes in this change stream.
+     */
+    public function __construct(ChangeStream $changeStream = null)
+    {
+        $this->changeStream = $changeStream;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getStack($path)
+    {
+        if (!$this->changeStream) {
```

Comment:
Shouldn't we rather rely on the parent implementation here, i.e. return a stack of size 1?